### PR TITLE
Add back in getattr call to Host methods

### PR DESCRIPTION
**Pull request by JacobCallahan**
These are still needed since self._session is stripped during the
pickling process.

### CODE REVIEW

File: broker/hosts.py [hunk 1]
```diff
@@ -29,7 +29,7 @@ def __del__(self):
 
     @property
     def session(self):
-        if not isinstance(self._session, Session):
+        if not isinstance(getattr(self, "_session", None), Session):
```
Comment (mshriver) [inline broker/hosts.py:32]:
```suggestion
        # This attribute may be missing after pickling
        if not isinstance(getattr(self, "_session", None), Session):
```

File: broker/hosts.py [hunk 2]
```diff
@@ -56,7 +56,7 @@ def connect(self, username=None, password=None):
         )
 
     def close(self):
-        if isinstance(self._session, Session):
+        if isinstance(getattr(self, "_session", None), Session):
```
Comment (mshriver) [inline broker/hosts.py:59]:
```suggestion
        # This attribute may be missing after pickling
        if isinstance(getattr(self, "_session", None), Session):
```

File: broker/hosts.py (hunk 1 above)
Comment (JacobCallahan) [reply broker/hosts.py:33]: done

File: broker/hosts.py (hunk 2 above)
Comment (JacobCallahan) [reply broker/hosts.py:61]: done

**Review by mshriver — APPROVED** [review]

**Pull request merged**